ext/dom/php_dom.stub.php

```diff
@@ -759,17 +759,77 @@ class DOMDocument extends DOM\Document
 
         public bool $substituteEntities;
 
+        /** @readonly */
+        public ?DOMElement $firstElementChild;
+
+        /** @readonly */
+        public ?DOMElement $lastElementChild;
+
+        /** @readonly */
+        public int $childElementCount;
+
         public function __construct(string $version = "1.0", string $encoding = "") {}
 
+        /** @return DOMAttr|false */
+        public function createAttribute(string $localName) {}
+
+        /** @return DOMAttr|false */
+        public function createAttributeNS(?string $namespace, string $qualifiedName) {}
+
+        /** @return DOMCdataSection|false */
+        public function createCDATASection(string $data) {}
```

Comment:
Reason why those are not proper return types?

Reply:
Looks like this is old DOM, it just looks like new code because I had to indent it, and this is because of BC reasons (class can be extended); and because this isn't conformant (see https://github.com/php/php-src/issues/12440#issuecomment-1763440275 where I asked the same question).

Reply:
Right